recipes/cpprestsdk/all/conanfile.py

```diff
@@ -0,0 +1,181 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class CppRestSDKConan(ConanFile):
+    name = "cpprestsdk"
+    description = "A project for cloud-based client-server communication in native code using a modern asynchronous " \
+                  "C++ API design"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Microsoft/cpprestsdk"
+    topics = ("conan", "cpprestsdk", "rest", "client", "http")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "exclude_websockets": [True, False],
+        "exclude_compression": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "exclude_websockets": False,
+        "exclude_compression": False
+    }
+    short_paths = True
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
+            del self.options.fPIC
```

Comment:
I'll drop this here...

I ran a [search](https://github.com/conan-io/conan-center-index/search?q=del+self.options.fPIC&unscoped_q=del+self.options.fPIC) to see how to deleted the fPIC option... However I have seen many different way is lots of recipes.

Which way is correct?

Reply:
Correct me if I'm wrong. @uilianries 
In `config_options()`, only the `settings` object is *finished*, so you can do:
```
if self.settings.os == "Windows":
    del self.options.fPIC
```
In the `configure()` method, the options object is finished, so you can do:
```
if self.options.shared:
    del self.options.fPIC
```

Reply:
@madebr Yes. Thus, it should be moved to config_options

Reply:
@uilianries I noticed this is very inconsistent in most recipes. Could this be added to the hooks?